### ISSUE TITLE
queryTimeoutUnit is not copied to DataModifyQuery in executeUpdate

### ISSUE THREAD

**Issue by sgflt** [eclipse-ee4j/eclipselink#1056]
Method DatabaseQuery#copyFromQuery is skipping field queryTimeoutUnit instead of copying it as stated in javadoc.

```java
    /**
     * INTERNAL: Copy all setting from the query. This is used to morph queries
     * from one type to the other. By default this calls prepareFromQuery, but
     * additional properties may be required to be copied as prepareFromQuery
     * only copies properties that affect the SQL.
     */
```

How to reproduce:
    query.setHint(QueryHints.QUERY_TIMEOUT, timeoutInSeconds);
    query.setHint(QueryHints.QUERY_TIMEOUT_UNIT, "SECONDS");

   query.executeUpdate(); // there are 180 milliseconds instead of seconds

**Comment by arjantijms** [comment]
@lukasj could you take a look at this? Would be nice to have fixed (it has been reported a long time ago already)

**Comment by lukasj** [comment]
fixed by #2161
